quinn-proto/src/tests/mod.rs

```diff
@@ -930,6 +930,28 @@ fn ack_delay_report() {
     );
 }
 
+#[test]
+fn connection_close_sends_acks() {
+    let _guard = subscribe();
+    let mut pair = Pair::default();
+    let (client_ch, _server_ch) = pair.connect();
+
+    let client_acks = pair.client_conn_mut(client_ch).stats().frame_rx.acks;
+
+    pair.client_conn_mut(client_ch).ping();
+    let time = pair.time;
+    pair.client_conn_mut(client_ch)
+        .close(time, VarInt(42), Bytes::new());
+
+    pair.drive();
+
+    let client_acks_2 = pair.client_conn_mut(client_ch).stats().frame_rx.acks;
+    assert!(
+        client_acks_2 > client_acks,
+        "Connection close should send pending ACKs"
+    );
```

Comment:
If the client's the one sending the close, shouldn't we check the server's received ACK count?

Reply:
Oh - yes and no - the server ACKs the clients ping, so we need to check whether the client got it.
But obviously we would need to close the server. Interesting that the test still captured the conditions :)

Reply:
fixed that by closing the server